org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/util/nodeselection/NodeSelection.java

```diff
@@ -74,7 +64,7 @@ public Object getFirstNode() {
      * @return the first node
      * @throws java.lang.IllegalStateException thrown if the selection is empty
      */
-    public <T> T getFirstNode(Class<T> expectedType) {
+    public <T> T getFirstElement(Class<T> expectedType) {
```

Comment:
Instead of renaming please just leave a todo there, it makes the PR unnecessarily big. This applies everywhere in this PR.